service/history/replication/ack_manager.go

```diff
@@ -686,3 +686,25 @@ func getVersionHistoryItems(
 	}
 	return versionhistory.CopyVersionHistory(versionHistoryBranch).GetItems(), versionHistoryBranch.GetBranchToken(), nil
 }
+
+func (p *ackMgrImpl) handleReadHistoryError(
+	namespaceID string,
```

Comment:
```suggestion
	namespaceID namespace.ID,
```
`namespace.ID` and `namespace.Name` should always be passed using dedicated type and `.String()` should be called in a very last moment.